wremnants/datasets/dataset_tools.py

```diff
@@ -119,15 +119,16 @@ def makeFilelist(paths, maxFiles=-1, base_path=None, nano_prod_tags=None, is_dat
 
         filelist.extend(files)
 
+    toreturn = filelist if maxFiles < 0 or len(filelist) < maxFiles else random.Random(1).sample(filelist, maxFiles)
```

Comment:
Why is this change needed ? I have the impression the result is the same

Reply:
without this fix the option oneMCfileEveryN does nothing